wallet/src/de/schildbach/wallet/ui/AbstractWalletActivity.java

```diff
@@ -18,37 +18,29 @@
 package de.schildbach.wallet.ui;
 
 import android.app.ActivityManager.TaskDescription;
-import android.os.Build;
 import android.os.Bundle;
 import android.view.View;
-import android.view.WindowManager;
 
 import androidx.annotation.LayoutRes;
 import androidx.appcompat.app.ActionBar;
 import androidx.appcompat.widget.Toolbar;
 
-import org.bitcoinj.wallet.Wallet;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import de.schildbach.wallet.WalletApplication;
 import de.schildbach.wallet_test.R;
 
 /**
  * @author Andreas Schildbach
  */
-public abstract class AbstractWalletActivity extends LockScreenActivity implements AbstractPINDialogFragment.WalletProvider {
-
-    private WalletApplication application;
+// TODO: do we need this class?
+public abstract class AbstractWalletActivity extends LockScreenActivity {
```

Comment:
perhaps DI and ViewModels make this class obsolete.  ViewModels should be the classes that do things with the application object rather than the Activity.